src/ctx.rs

```diff
@@ -99,6 +99,32 @@ impl<const UP_TO: usize, const WITH_DEFAULT: bool> CaseSetter<UP_TO, WITH_DEFAUL
         let mut buf = buf.index_mut(self.offset..self.offset + self.len);
         small_memset::<V, UP_TO, WITH_DEFAULT>(&mut *buf, val);
     }
+
+    /// # Safety
+    /// 
+    /// `buf` must be correctly aligned and dereferencable (but need not be
+    /// initialized).
+    #[inline]
+    pub unsafe fn set_raw<T: Clone + Copy>(&self, buf: *mut [T], val: T) {
```

Comment:
Could you have this just call `small_memset_raw` just so all the small memset code is next to each other?

Reply:
I'd prefer not to because I don't want to actually subslice the `buf` slice as we do in the other cases.

Reply:
I still think it's easier to read with them next to each other even if we don't subslice `buf`.  It's just helpful to have them visually next to each other.